### PR TITLE
added tests and functionality for getting single article

### PR DESCRIPTION
**Pull request by SoheylNabati**

### CODE REVIEW

File: models/newsModels.js [hunk 1]
```diff
@@ -5,3 +5,11 @@ exports.fetchTopics = () => {
     return rows;
   });
 };
+
+exports.fetchArticle = (id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id=$1;`, [id])
+    .then(({ rows }) => {
+      return rows[0];
+    });
```
Comment (two-triangles) [inline models/newsModels.js:17]: There is currently no error handling for this endpoint. How would your server reply if there was an id number that was valid, but but no article exists with that id number? Or how would your server respond if the provided id number was of an invalid data type - like /api/articles/cheese for example?

File: controllers/newsControllers.js [hunk 2]
```diff
@@ -1,6 +1,15 @@
-const { fetchTopics } = require("../models/newsModels");
+const { fetchTopics, fetchArticle } = require("../models/newsModels");
 exports.getTopics = (req, res, next) => {
   return fetchTopics().then((topics) => {
     return res.status(200).send({ topics });
   });
 };
+
+exports.getArticle = (req, res, next) => {
```
Comment (two-triangles) [inline controllers/newsControllers.js]: I think getArticleById is a more semantic name for this function. This is not an urgent fix, but just something to consider when naming your functions

**Comment by two-triangles** [comment]
Please implement error handling for this endpoint and send me a message when you have done so. I will have a look over before approving merge :)

**Comment by two-triangles** [comment]
I am happy for you to merge this branch. Once you have merged this, please pull main into your branch for task 4, before carrying on with the recommended amendments on that branch. This was your branch for task 4 will be up to date with the status of the current remove main branch

File: models/newsModels.js (hunk 1 above)
Comment (SoheylNabati) [reply models/newsModels.js:17]: added error handling and changed the name of the function could you have a look please 
thank you

File: __tests__/app.test.js [hunk 3]
```diff
@@ -43,3 +43,39 @@ describe("app", () => {
...
+
+describe.only("GET /api/articles/:article_id", () => {
+  it("should respond with a single article object, which should have author property which is the username from the users table, title, article_id, body, topic created_at and votes property with a status code of 200", () => {
+    const article_id = 2;
+    return request(app)
+      .get(`/api/articles/${article_id}`)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toEqual({
+          article_id: article_id,
```
Comment (itskatherine) [inline __tests__/app.test.js:55]: I like how you've using a variable here to check the article ids match 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -43,3 +43,39 @@ describe("app", () => {
...
+          created_at: "2020-10-16T05:03:00.000Z",
+          votes: 0,
+        });
+      });
+  });
+  it("400: respond with an error message when pased an article id that is invalid type", () => {
+    return request(app)
+      .get(`/api/articles/cheese`)
```
Comment (itskatherine) [inline __tests__/app.test.js:67]: Great now I'm hungry

File: app.js [hunk 5]
```diff
@@ -1,15 +1,21 @@
 const express = require("express");
-const { getTopics } = require("./controllers/newsControllers");
+const { getTopics, getArticleById } = require("./controllers/newsControllers");
+const {
+  invalidEndpoint,
+  serverError,
+  PSQLerrors,
+  costumError,
+} = require(`./controllers/errorHandling.controller`);
```
Comment (itskatherine) [inline app.js:8]: Awesome work separating out into their own error controller, one small thing, it's spelled "custom" errors

**Review by itskatherine — COMMENTED** [review]
Hey Sol, some minor things, have a look through then feel free to merge 👍

**Pull request merged**